common/static/common/js/discussion/utils.js

```diff
@@ -386,26 +410,28 @@
                         return $3;
                     });
                 } else {
-                    processedText += text;
+                    processedText += safeHtml;
                     break;
                 }
             }
             /* jshint +W083 */
-            text = processedText;
-            text = text.replace(new RegExp(ESCAPED_DOLLAR, 'g'), '\\$');
-            text = text.replace(/\\\\\\\\/g, ESCAPED_BACKSLASH);
-            text = text.replace(/\\begin\{([a-z]*\*?)\}([\s\S]*?)\\end\{\1\}/img, function($0, $1, $2) {
+            safeHtml = processedText;
+            safeHtml = safeHtml.replace(new RegExp(ESCAPED_DOLLAR, 'g'), '\\$');
+            safeHtml = safeHtml.replace(/\\\\\\\\/g, ESCAPED_BACKSLASH);
+            safeHtml = safeHtml.replace(/\\begin\{([a-z]*\*?)\}([\s\S]*?)\\end\{\1\}/img, function($0, $1, $2) {
                 return processor(("\\begin{" + $1 + "}") + $2 + ("\\end{" + $1 + "}"));
             });
-            text = text.replace(new RegExp(ESCAPED_BACKSLASH, 'g'), '\\\\\\\\');
-            $div = $("<div>").html(text);
+            safeHtml = safeHtml.replace(new RegExp(ESCAPED_BACKSLASH, 'g'), '\\\\\\\\');
+            // safe-lint: disable=javascript-jquery-html
+            $div = $("<div>").html(safeHtml);
             cnt = 0;
             $div.find("code").each(function(index, code) {
+                // safe-lint: disable=javascript-jquery-html
                 $(code).html(processor(codeArchive[cnt], 'code'));
```

Comment:
I'd prefer seeing any of the many options other than disabling the rule.  
1. Could `processor(codeArchive[cnt], 'code')` return an HTML Snippet (using HtmlUtils)?  This is typically the best option if possible.
2. If this is not possible, I prefer `HtmlUtils.HTML(processor(codeArchive[cnt], 'code'))` to declare this as HTML rather than disabling the linter.  This basically does the same thing, which is why option #1 is better, but I prefer seeing the solution in code than in a comment where possible.